### DISCUSSION TITLE
Combi mode: run directly for drun, open terminal emulator for run

### DISCUSSION THREAD

**Discussion by pacien** [davatorium/rofi#1793] [Q&A] [answered]
I wish to have both desktop entries and executable in the PATH listed in rofi.
As suggested by the manual page, I'm using:
`rofi -show combi -modes combi -combi-modes 'drun,run'`

This launches desktop entries from the drun mode properly,
but does not start executables from the run mode in a terminal emulator.

Setting `-run-command '{terminal} -e {cmd}'` fixes running run mode entries,
but it also makes drun mode entries start in a new terminal emulator.

Is there a way to make, in combi mode:
- drun mode entries start directly (`{cmd}`), and at the same time
- run mode entries start in a terminal emulator (`{terminal} -e {cmd}`)?

I would like this to be automatic and not having to use `Shift-Enter` manually
for some entries, which is too error-prone.

**Comment by DaveDavenport** [comment, marked as answer]
No,  there is no way to do this.   Don't think I got this request before.

Launching in a terminal depends on application. In drun mode the desktop file determines this. In run mode their is no way to determine it and it is left to the user (by using the right binding).
Given run mode also lists applications that have their own UI, I find it odd to always launch in terminal.

**Reply by pacien** [reply]
Thanks for the answer!

My use case for the run mode is mainly to launch REPLs and TUIs applications
which do not usually have desktop files.

But yes, it makes more sense to define the missing desktop files to know when
to launch a terminal emulator or not.

**Reply by DaveDavenport** [reply]
There used to be a nice program to create desktop files and to tweak them. I used it years ago but cannot find it back anymore (pretty frustrating).
Maybe we should make one..
